### PR TITLE
Updated few FormatFilter tests to include Accept header

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/FormatFilterSampleTest.cs [hunk 1]
```diff
@@ -42,8 +43,15 @@ public FormatFilterSampleTest(MvcTestFixture<FormatFilterSample.Web.Startup> fix
         [Fact]
         public async Task FormatFilter_ExtensionInRequest_Optional()
         {
-            // Arrange & Act
-            var response = await Client.GetAsync("http://localhost/FormatFilter/GetProduct.json");
+            // Arrange
+            var request = new HttpRequestMessage();
+            // Typical accept header sent by Chrome browser
+            request.Headers.Add("Accept", "text/html,application/xhtml+xml," +
+                "application/xml;q=0.9,image/webp,*/*;q=0.8");
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/FormatFilterSampleTest.cs]: Please move the comment and the `string` to a `const`. Just use the `const` here and in the other tests.

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/FormatFilterSampleTest.cs [hunk 2]
```diff
@@ -42,8 +43,15 @@ public FormatFilterSampleTest(MvcTestFixture<FormatFilterSample.Web.Startup> fix
         [Fact]
         public async Task FormatFilter_ExtensionInRequest_Optional()
         {
-            // Arrange & Act
-            var response = await Client.GetAsync("http://localhost/FormatFilter/GetProduct.json");
+            // Arrange
+            var request = new HttpRequestMessage();
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/FormatFilterSampleTest.cs]: Be more explicit and shorten the setup to match most other `HttpRequestMessage`s used in tests: `new HttpRequestMessage(HttpMethod.Get, "http://localhost/FormatFilter/GetProduct.json")`.

Then remove [line 51](https://github.com/aspnet/Mvc/pull/4034/files#diff-59701ed5c13c8176b640aaa41f872c67R51).

**Comment by dougbu** [comment]
:watch:

**Comment by kichalla** [comment]
:up: :date:

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**